content/08-mongodb/10-mongodb-transactions.mdx

```diff
@@ -0,0 +1,246 @@
+  {
+    _id: ObjectId("620397dd4b871fc65c193106"),
+    first_name: 'James',
+    last_name: 'Joyce',
+    best_title: 'Ulysses'
+  },
+  {
+    _id: ObjectId("620398016ed0bb9e23785973"),
+    first_name: 'William',
+    last_name: 'Gibson',
+    best_title: 'Neuromancer'
+  },
+  {
+    _id: ObjectId("6203981d6ed0bb9e23785974"),
+    first_name: 'George',
+    last_name: 'Orwell',
+    best_title: 'Homage to Catalonia'
+  },
+  {
+    _id: ObjectId("620398516ed0bb9e23785975"),
+    first_name: 'James',
+    last_name: 'Baldwin',
+    best_title: 'The Fire Next Time'
+  },
+  {
+    _id: ObjectId("6203a075c374636bc6976baa"),
+    first_name: 'Virginie',
+    last_name: 'Despentes',
+    best_title: 'Vernon Subutex'
+  }
+]
+```
+
+Since this transaction session has still not been committed, we will not see it return the same result in a shell existing outside of the session. To confirm, we run `db.authors.find()` in another shell instance and we return the original documents.
+
+```
+[
+  {
+    _id: ObjectId("620397dd4b871fc65c193106"),
+    first_name: 'James',
+    last_name: 'Joyce',
+    best_title: 'Ulysses'
+  },
+  {
+    _id: ObjectId("620398016ed0bb9e23785973"),
+    first_name: 'William',
+    last_name: 'Gibson',
+    best_title: 'Neuromancer'
+  },
+  {
+    _id: ObjectId("6203981d6ed0bb9e23785974"),
+    first_name: 'George',
+    last_name: 'Orwell',
+    best_title: 'Homage to Catalonia'
+  },
+  {
+    _id: ObjectId("620398516ed0bb9e23785975"),
+    first_name: 'James',
+    last_name: 'Baldwin',
+    best_title: 'The Fire Next Time'
+  }
+]
+```
+This difference demonstrates how the transaction is still currently in a state of ambiguity. It can still either succeed and be committed to the database or it could fail. Either way the database will come back to a consistent state of either including the new record or remaining the same as before the transaction session started.
+
+To commit the transaction and bring the database to the new consistent state witht the new author, we run the following in the session context:
+
+```
+session.commitTransaction()
+```  
+Now, the new record will exist in both instances, the shell where the transaction session was taking place as well as any shell outside of the transaction.
+  
+
+### Aborting a transaction
+
+Now that we have gone through starting a session and working all the way to committing it, we can explore the alternative result for a transaction.
+
+This path begins the same and changes only at the commit point. If we want to discard the changes the transaction is making, then we use the `abortTransaction()` method as follows in the session shell:
+
+```
+session.abortTransaction()
+```
+This method cancels the transaction session and discards any potential changes. The database returns to its original state. Instead of resulting in a new consistent state, the database rolls back the changes and stays the same as when the transaction session first started.  
+
+## Conclusion
+
```

Comment:
These two paragraphs in the conclusion could probably be swapped if you want.  I'm not sure if it makes more sense to start with the summary of the article's topics or to start with why transactions are important, so I just wanted to mention that it would work both ways so you can decide what you like better.